### PR TITLE
Implementing e2e tests on iOS

### PR DESCRIPTION
**Pull request by hgvo9**
### Checklist
✅ TDD Security.
✅ Minimal Testing.
✅ Accurate assignees / reviewers.
- - -
### Description
This PR is focused on the e2e tests on iOS that I have implemented thanks to Detox which makes it possible (https://github.com/wix/Detox)

For those who don't know, e2e tests are made to only test the application on the front-end and to not care about what happens in the code. This is also called gray box testing, you're testing the product (application mobile in our case) with little knowledge about the structure of the application.

Basically, with this PR merged, we will have e2e tests that can run locally on any iPhone devices (simulator) to test differents scenarios as :

https://user-images.githubusercontent.com/62654884/124925938-4870bd00-dffd-11eb-847d-d7655a99e7d1.mp4

In addition to that, at every push, a github action will be executing these e2e tests in a macos environment on an simulated iPhone.

image

### CODE REVIEW

File: .github/workflows/ios_e2e_tests.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+        
+    - name: Create env file
+      run: |
+        touch .env
+        echo API_URL=${{ secrets.API_URL }} >> .env
```
Comment (CptHash) [inline .github/workflows/ios_e2e_tests.yml:21]: Are you using the production API for your testing ?

**Review by CptHash — APPROVED** [review]
Really happy to see this kind of test ! It seems easy to scale :rocket:

File: .github/workflows/ios_e2e_tests.yml (hunk 1 above)
Comment (hgvo9) [reply .github/workflows/ios_e2e_tests.yml:21]: Yes, I'm using this one 'https://api.safely-app.fr', is that okay ?

Comment (CptHash) [reply .github/workflows/ios_e2e_tests.yml:21]: Yes perfect :+1:

**Comment by louischristner** [comment]
I did not test it already but for the moment it looks good to me, I will be giving you more feedback after testing. Great job nonetheless !

File: android/app/src/main/java/com/safelymobileapp/MainApplication.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import com.facebook.react.ReactApplication;
 import com.oblador.vectoricons.VectorIconsPackage;
 import com.oblador.vectoricons.VectorIconsPackage;
+import com.oblador.vectoricons.VectorIconsPackage;
```
Comment (Trushy) [inline android/app/src/main/java/com/safelymobileapp/MainApplication.java:9]: Why do you 3 times the same import ? Is this necessarry or I am missing on something ?

Comment (Trushy) [reply android/app/src/main/java/com/safelymobileapp/MainApplication.java:9]: Is this a generated file ?

File: android/settings.gradle [hunk 3]
```diff
@@ -3,5 +3,7 @@ include ':react-native-vector-icons'
 project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
 include ':react-native-vector-icons'
 project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
+include ':react-native-vector-icons'
```
Comment (Trushy) [inline android/settings.gradle:6]: Same

**Review by Trushy — APPROVED** [review]

File: android/app/src/main/java/com/safelymobileapp/MainApplication.java (hunk 2 above)
Comment (hgvo9) [reply android/app/src/main/java/com/safelymobileapp/MainApplication.java]: Thanks for noticing it, https://github.com/safely-app/Front-End-Mobile/pull/14/commits/70a0bf17e340d26abb15622750b38426a284ba68

File: android/settings.gradle (hunk 3 above)
Comment (hgvo9) [reply android/settings.gradle]: Thanks for noticing it, https://github.com/safely-app/Front-End-Mobile/pull/14/commits/70a0bf17e340d26abb15622750b38426a284ba68

**Review by louischristner — APPROVED** [review]
I am very sad that I am not able to test it :(
Apart from that I approved. Great job !
